idb/postgres/internal/encoding/encode.go

```diff
@@ -63,13 +127,123 @@ func EncodeSignedTxnWithAD(stxn types.SignedTxnWithAD) []byte {
 	return EncodeJSON(convertSignedTxnWithAD(stxn))
 }
 
-func init() {
-	jsonCodecHandle = new(codec.JsonHandle)
-	jsonCodecHandle.ErrorIfNoField = true
-	jsonCodecHandle.ErrorIfNoArrayExpand = true
-	jsonCodecHandle.Canonical = true
-	jsonCodecHandle.RecursiveEmptyCheck = true
-	jsonCodecHandle.HTMLCharsAsIs = true
-	jsonCodecHandle.Indent = 0
-	jsonCodecHandle.MapKeyAsString = true
+func EscapeNulls(x string) string {
+	xb := []byte(x)
+	newlen := 0
+	i := 0
+	for i < len(xb) {
+		c, csize := utf8.DecodeRune(xb[i:])
+		if c == utf8.RuneError {
+			newlen += 6 * csize // \uxxxx
+			i += csize
+			continue
+		}
+		// TODO: unicode.IsPrint(c) ?
+		switch c {
+		case 0:
+			newlen += 6 // \u0000
+		case '\\':
+			newlen += 2
+		default:
+			newlen+= csize
+		}
+		i += csize
+
+	}
+	if len(x) == newlen {
+		return x
+	}
+	const escapenull = "\\u0000"
+	out := make([]byte, newlen)
+	var runehex = [6]byte{'\\', 'u', '0', '0'}
+	start := 0
+	outpos := 0
+	i = 0
+	for i < len(xb) {
+		c, csize := utf8.DecodeRune(xb[i:])
+		if c == utf8.RuneError {
+			copy(out[outpos:], xb[start:i])
+			outpos += i - start
+			start = i + csize
+			for xi := i; xi < start; xi++ {
+				hex.Encode(runehex[4:], xb[xi:xi+1])
+				copy(out[outpos:], runehex[:])
+				outpos += 6
+			}
+			i += csize
+			continue
+		}
+		switch c {
+		case 0:
+			copy(out[outpos:], xb[start:i])
+			outpos += i - start
+			start = i + 1
+			copy(out[outpos:], escapenull)
+			outpos += 6
+		case '\\':
+			copy(out[outpos:], xb[start:i])
+			outpos += i - start
+			start = i + 1
+			out[outpos] = '\\'
+			outpos++
+			out[outpos] = '\\'
+			outpos++
+		default:
+		}
+		i += csize
+	}
+	if start < len(xb) {
+		copy(out[outpos:], xb[start:])
+	}
+	return string(out)
+}
+
+// UnescapeNulls is the inverse function of EscapeNulls.
+// UnescapeNulls converts \\ and \uXXXX back into their unescaped form but may not be fully general for input not generated by EscapeNulls().
```

Comment:
I believe the method is correct - but relying on the fact that it was correctly generated by EscapeNulls() is an issue. It need to be refactored to return an error in case it's not possible to unescape.

Reply:
If this is all only used internally to store a few fields and read them back, then there is no problem?